### PR TITLE
storage: fix blob batch read error

### PR DESCRIPTION
**Pull request by kevinXYin**
this pr will fix https://github.com/dragonflyoss/image-service/issues/693

### CODE REVIEW

**Review by hsiangkao — APPROVED** [review]

File: storage/src/meta/mod.rs [hunk 1]
```diff
@@ -489,6 +489,9 @@ impl BlobMetaInfo {
                 }
             }
 
+            if last_end > end {
```
Comment (changweige) [inline storage/src/meta/mod.rs:492]: Should compare `last_end` with `batch_end` ? And `last_end` == `last_end` should be valid

Comment (hsiangkao) [reply storage/src/meta/mod.rs:492]: let mut last_end = entry.aligned_uncompressed_end();

Comment (hsiangkao) [reply storage/src/meta/mod.rs:492]: @kevinXYin

Comment (kevinXYin) [reply storage/src/meta/mod.rs:492]: in while loop already compare last_end with batch_end , for https://github.com/dragonflyoss/image-service/issues/693 issue last chunk's aligned_uncompressed_end not reach the batch_end which is state.uncompressed_size. And I am not sure is this valid (state.uncompressed_size > last chunk's aligned_uncompressed_end) ? But I think just checking the last_end to cover the end is enough.  And , yes last_end == end should be valid , I will send another pr to change to `if last_end >= end` , does this make sense ? @changweige @hsiangkao

Comment (changweige) [reply storage/src/meta/mod.rs:492]: The root cause of this problem is blob's uncompressed size is wrongly calculated. I have raised a PR to fix the image format. https://github.com/dragonflyoss/image-service/pull/712

**Pull request merged**